### PR TITLE
USHIFT-1620: add a script to produce the full rebase history

### PR DESCRIPTION
**Pull request by dhellmann**
This script is useful for building the technical enablement summary
and release notes.

/assign @pmtk @pacevedom

### CODE REVIEW

File: scripts/auto-rebase/rebase_history.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+#
+# This script is a tool for determining what changes came in via rebases for a release.
+
+set -euo pipefail
+
+# Find the point where the release branch diverged from the main branch.
+branch_start() {
+    local -r branch_name="$1"
+
+	git show --pretty=format:%H "$(git rev-list "$(git rev-list --first-parent ^"${branch_name}" origin/main | tail -n1)^^!")"
```
Comment (pmtk) [inline scripts/auto-rebase/rebase_history.sh:11]:
```suggestion
	git show --pretty=format:%H "$(git rev-list "$(git rev-list --first-parent ^"origin/${branch_name}" origin/main | tail -n1)^^!")"
```

I didn't have `release-4.13` branch in my local repo:

```
bash -x scripts/auto-rebase/rebase_history.sh                                                                                                                                                                                                                         08/25/2023 11:43:42 AM
+ set -euo pipefail
++ awk '{print $3}' Makefile.version.aarch64.var
++ cut -f1 -d-
+ SOURCE_VERSION=4.14.0
++ echo 4.14.0
++ cut -f2 -d.
+ MINOR_VERSION=14
+ PREVIOUS_MINOR_VERSION=13
+ PREVIOUS_RELEASE_BRANCH=release-4.13
++ branch_start release-4.13
++ local -r branch_name=release-4.13
++++ git rev-list --first-parent '^release-4.13' origin/main
++++ tail -n1
fatal: bad revision '^release-4.13'
+++ git rev-list '^^!'
fatal: bad revision '^^!'
++ git show --pretty=format:%H ''
fatal: ambiguous argument '': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
+ FIRST_RELEASE_COMMIT=
```

Comment (dhellmann) [reply scripts/auto-rebase/rebase_history.sh:11]: d'oh, fixed

Comment (dhellmann) [reply scripts/auto-rebase/rebase_history.sh:11]: I fixed this on line 28 by updating the name of the branch in the variable there.

**Pull request merged**